server/controllers/profileController.js

```diff
@@ -0,0 +1,46 @@
+import User from '@models/User';
+
+const getUser = async id => {
+    const {
+        firstName,
+        lastName,
+        email,
+        bio,
+        avatar
+    } = await User.query().findById(id);
+
+    return { firstName, lastName, email, bio, avatar };
+};
+
+const getProfile = async (req, res) => {
+    const userId = req.params.id;
+    const thisUser = req.user;
+
+    const id = userId ? userId : thisUser.id;
+
+    const user = await getUser(id);
+    return res.status(200).jsend(user);
+};
+
+const editProfile = async (req, res) => {
+    const { id } = req.user;
+
+    const { firstName, lastName, email, bio, avatar } = req.body;
+
+    const update = async (column, value) => {
+        await User.query()
+            .findById(id)
+            .patch({ [column]: value });
+    };
+
+    if (firstName) await update('firstName', firstName);
```

Comment:
This is performing 5 database in one request if all data is provided? It's not needed. You can do this

```js
await User.query().findById(id).patch({
 firstName: firstName || req.user.firstName,
 lastName: lastName || req.user.lastName
})
```

With this, no matter what fields are provided, only one database call is made. Also, you can write a function that filters out undefined data from `req.body` so only defined fields are patched.

Reply:
ok